### DISCUSSION TITLE
Android SMS Gateway PHP API Client

### DISCUSSION THREAD

**Discussion by dvxgit-tbautista** [capcom6/android-sms-gateway#88] [Q&A] [answered]
Hello,

I recently tried using the Android SMS Gateway PHP API Client and encountered an error. Could you please assist me with this issue?

The error message is as follows:

Error: Class 'Http\Client\Exception\HttpException' not found in C:\wamp64\www\capcomPHP\vendor\capcom6\android-sms-gateway\src\Client.php on line 119

Thank you for your help.

Best regards,
Diane

**Comment by capcom6** [comment]
Hello @dvxgit-tbautista ,

It seems like there might be an issue with missing dependencies. Could you please provide additional details such as your PHP version and the versions of any relevant packages? You can obtain this information by running `composer show | grep php-http` in your terminal.

Thank you for reaching out.

Best regards,
Aleksandr

**Reply by dvxgit-tbautista** [reply]
Hello Aleksandr,

I appreciate you taking the time to reply to my concern. The PHP version I am currently using is 7.4.

Here is the screenshot of the packages:
![packages](https://github.com/capcom6/android-sms-gateway/assets/110645703/b0dd5443-e47f-45fe-9677-2568746dbf6e)

**Reply by capcom6** [reply]
I've installed the same dependencies on my end, and everything is functioning correctly. Are you developing your application from scratch, or are you utilizing a specific framework? To better assist you, could you provide a minimal example of your code that reproduces the issue?

I conducted a test with the following code:

```php title="src/test.php"
<?php

require __DIR__ . '/../vendor/autoload.php';

use AndroidSmsGateway\Client;
use AndroidSmsGateway\Domain\Message;

$login = 'LOGIN';
$password = 'PASSWORD'; // Ensure you replace 'PASSWORD' with your actual password

$client = new Client($login, $password);

$message = new Message('Your message text here.', ['+12345678901']);

try {
    $messageState = $client->send($message);
    // Uncomment the next line to print the message state
    // print_r($messageState);
    echo "Message ID: " . $messageState->getId() . PHP_EOL;
} catch (Exception $e) {
    echo "Error sending message: " . $e->getMessage() . PHP_EOL;
    die(1);
}
```

**Reply by dvxgit-tbautista** [reply]
I haven't started developing the app; I've only tried your repository and followed your README here.

https://github.com/capcom6/android-sms-gateway-php

**Reply by dvxgit-tbautista** [reply]
Here's my screenshot

Maybe there's a configuration I forgot to configure?
![image](https://github.com/capcom6/android-sms-gateway/assets/110645703/3992ffd3-e569-48ba-b48e-116be2d6ed43)

**Reply by capcom6** [reply]
Please try reinstalling the dependencies to ensure that everything is installed correctly. You can follow these steps:

```sh
rm -rf vendor/
rm -f composer.lock
composer require capcom6/android-sms-gateway
php index.php
```

Alternatively, you might consider recreating the project entirely with:

```sh
rm -rf vendor/
rm -f composer.lock composer.json
composer init -n  # The `-n` option will skip the interactive prompts
composer require capcom6/android-sms-gateway
php index.php
```

This issue might arise from classes not autoloading properly. I have already tested this code on Windows using PowerShell, and everything seemed to be fine.

**Reply by dvxgit-tbautista** [reply]
Hi, I've just tried your suggestions, but I still get the same error. Are the login and password the ones in the 'Authorization' section? Where do I place my local address here?

**Reply by capcom6** [reply]
This issue is not related to the specific logic of the SMS gateway client; rather, it appears to be a problem with Composer's autoloading or the PHP configuration on your computer. Unfortunately, without direct access to your system, it's difficult for me to suggest any additional steps to remedy this. I recommend trying to run the same code in a Docker container or on any Linux-based system to mitigate potential local system configuration issues.

Regarding your question about login credentials, yes, the login and password are the ones specified in the "Authorization" section. The server address should be passed as the third argument to the `Client` class constructor.

**Reply by dvxgit-tbautista** [reply]
Will you take a look at this

![image](https://github.com/capcom6/android-sms-gateway/assets/110645703/075c2379-2fcf-48ab-a385-c16304a6b007)

**Reply by dvxgit-tbautista** [reply]
Hey @capcom6 

I have finally made it work. I needed to change the DEFAULT_URL in the Client.php.

**Reply by capcom6** [reply]
To clarify, the server address should be passed as the **third** argument, not the fourth as on your screenshot.

Modifying the library code directly on your local machine is generally not advisable. Such changes are temporary and will be overwritten during the next update of the library.

Additionally, could you share what steps you took to resolve the initial issue regarding the 'Class not found' error?

**Reply by dvxgit-tbautista** [reply]
Here are the steps I did when I encountered the initial issue:

1. I cloned your repository.
2. Changed the login and password.
3. Changed the number.
4. Ran php index.php.

Edit again: The $login and $password are the only ones I modified at first.

![image](https://github.com/capcom6/android-sms-gateway/assets/110645703/022c4315-7362-49ae-b72a-00cd16c85b22)

**Reply by dvxgit-tbautista** [reply]
When I changed this one, it worked and the error no longer showed.

![image](https://github.com/capcom6/android-sms-gateway/assets/110645703/f3ceebd5-8578-4b72-bd72-185d464dc704)

Edit: When I changed it to my Local Address in the SmsGateway app.

**Reply by capcom6** [reply, marked as answer]
I believe there might be a misunderstanding in this sequence of events. The modification of the `DEFAULT_URL` line does not directly relate to the initial "Class not found" error. As previously mentioned, there's no need to edit the library code directly; you can simply pass the address as the appropriate parameter in the constructor.

Nonetheless, if your issue has been resolved, I'm pleased to hear it.

**Reply by dvxgit-tbautista** [reply]
Alright, I get it now.

I did not change anything in the library this time and only passed three arguments in the Client: login, password, and the address.

Thank you so much for the help Aleksandr. I apologize for taking up too much of your time.
